src/v/raft/consensus.h

```diff
@@ -845,15 +845,22 @@ class consensus {
     /// used to wait for background ops before shutting down
     ss::gate _bg;
 
+    /**
+     * Locks listed in the order of nestedness, election being the outermost
+     * and snapshot the innermost. I.e. if any of these locks are used at the
+     * same time, they should be acquired in the listed order and released in
+     * reverse order.
+     */
+    /// guards from concurrent election where this instance is a candidate
+    mutex _election_lock{"consensus::election_lock"};
```

Comment:
do we need to mark this as broken() at shutdown

Reply:
we probably do to speed up shutdown, thanks
done